### PR TITLE
BN-1246-update-proposal-validation

### PR DESCRIPTION
**Pull request by nandotorterolo**
## Purpose
- BN-1246-update-proposal-validation 

## Approach
-  new validation rule for proposal updated, simple rules, all values should be positives

## Testing
- unit test

## Tickets
BN-1246

### CODE REVIEW

File: brambl-sdk/src/main/scala/co/topl/brambl/validation/TransactionSyntaxInterpreter.scala [hunk 1]
```diff
@@ -569,4 +570,25 @@ object TransactionSyntaxInterpreter {
     )
 
   }
+
+  private def updateProposalValidation(transaction: IoTransaction) = {
+    val upsIn = transaction.inputs.filter(_.value.value.isUpdateProposal).map(_.value.getUpdateProposal)
```
Comment (SeanCheatham) [inline brambl-sdk/src/main/scala/co/topl/brambl/validation/TransactionSyntaxInterpreter.scala:575]: You can probably skip the validation of the inputs since they would have been validated already by the previous transaction.

File: brambl-sdk/src/main/scala/co/topl/brambl/validation/TransactionSyntaxInterpreter.scala [hunk 2]
```diff
@@ -569,4 +570,25 @@ object TransactionSyntaxInterpreter {
...
+    val upsOut = transaction.outputs.filter(_.value.value.isUpdateProposal).map(_.value.getUpdateProposal)
```
Comment (SeanCheatham) [inline brambl-sdk/src/main/scala/co/topl/brambl/validation/TransactionSyntaxInterpreter.scala:576]: I think you can shorten this to `transaction.outputs.flatMap(_.value.value.updateProposal)`

**Review by SeanCheatham — APPROVED** [review]

**Review by DiademShoukralla — APPROVED** [review]

File: brambl-sdk/src/main/scala/co/topl/brambl/validation/TransactionSyntaxInterpreter.scala (hunk 1 above)
Comment (nandotorterolo) [reply brambl-sdk/src/main/scala/co/topl/brambl/validation/TransactionSyntaxInterpreter.scala:575]: thanks for the suggestion

File: brambl-sdk/src/main/scala/co/topl/brambl/validation/TransactionSyntaxInterpreter.scala (hunk 2 above)
Comment (nandotorterolo) [reply brambl-sdk/src/main/scala/co/topl/brambl/validation/TransactionSyntaxInterpreter.scala:576]: thanks for the suggestion
